MainWindow.cpp

```diff
@@ -265,7 +265,19 @@ MainWindow::MessageReceived(BMessage* message)
 				if (title == contents)
 					title = "";
 
-				fFavorites->AddItem(new FavItem(contents, title, index + 1));
+				bool isDuplicate = false;
+				for (int32 i = 0; i < index; ++i) {
+					FavItem* item
+						= static_cast<FavItem*>(fFavorites->ItemAt(i));
+					if (item->GetClip() == contents) {
+						isDuplicate = true;
+						break;
+					}
+				}
+
+				if (!isDuplicate)
+					fFavorites->AddItem(
+						new FavItem(contents, title, index + 1));
```

Comment:
If it is a duplicate, we should 'break' here. Otherwise the rest of the case block is run, and we can end up re-ordering the list of favourites.